src/layers/sds/cpm_translation/modify_key.py

```diff
@@ -1,186 +1,186 @@
-    for device in devices:
-        yield device
-    yield new_device
-
-
-def replace_accredited_systems(
-    devices: list[Device], field_name: str, value: str
-) -> Generator[Device, None, None]:
-    current_ods_codes = {device.ods_code for device in devices}
-    final_ods_codes = NhsAccreditedSystem.parse_and_validate_field(
-        field=field_name, value=value
-    )
-    removed_ods_codes = current_ods_codes - final_ods_codes
-    for device in devices:
-        if device.ods_code in removed_ods_codes:
-            device.delete()
-        yield device
-
-    for new_ods_code in final_ods_codes - current_ods_codes:
-        *_, new_device = copy_new_accredited_system_from_sibling_device(
-            devices=devices, field_name=field_name, value=[new_ods_code]
-        )
-        yield new_device
-
-
-def _get_msg_handling_system_key(responses: list[dict]) -> MessageHandlingSystemKey:
-    """Construct the MHS scoped party key from questionnaire responses"""
-    return MessageHandlingSystemKey(
-        **{
-            key: values
-            for key in NhsMhs.key_fields()
-            for values in get_in_list_of_dict(obj=responses, key=key)
-        }
-    )
-
-
-def replace_msg_handling_system(
-    devices: list[Device], field_name: str, value: str
-) -> Generator[Device, None, None]:
-    (device,) = devices
-
-    (questionnaire_response_by_datetime,) = device.questionnaire_responses.values()
-    (_questionnaire_response,) = questionnaire_response_by_datetime.values()
-    _questionnaire_response.questionnaire = NhsMhs.questionnaire()
-
-    new_value = NhsMhs.parse_and_validate_field(field=field_name, value=value)
-
-    # Nothing to do if replacing with itself
-    old_value = _questionnaire_response.flat_answers[field_name]
-    if new_value == old_value:
-        return
-    device.delete()
-    yield device
-
-    questionnaire_response = new_questionnaire_response_from_template(
-        questionnaire_response=_questionnaire_response,
-        field_to_update=field_name,
-        value=new_value,
-    )
-    msg_handling_system_key = _get_msg_handling_system_key(
-        responses=_questionnaire_response.answers
-    )
-    new_scoped_party_key = DEVICE_KEY_SEPARATOR.join(astuple(msg_handling_system_key))
-    product_team = ProductTeam(
-        id=device.product_team_id,
-        ods_code=msg_handling_system_key.nhs_id_code,
-        name=DEFAULT_PRODUCT_TEAM["name"],
-    )
-    new_device = product_team.create_device(
-        name=device.name, device_type=device.device_type
-    )
-    new_device.add_questionnaire_response(questionnaire_response=questionnaire_response)
-    new_device.add_key(
-        key_type=DeviceKeyType.MESSAGE_HANDLING_SYSTEM_ID,
-        key_value=new_scoped_party_key,
-    )
-    set_device_tags(
-        device=new_device,
-        data=questionnaire_response.flat_answers,
-        model=SearchSDSEndpointQueryParams,
-    )
-
-    yield new_device
+# from dataclasses import astuple
```

Comment:
Comments ???

Reply:
ah this file is getting deleted soon and is getting used anywhere but could be useful for reference when rewriting the changelog